utils/utils.go

```diff
@@ -108,4 +115,46 @@ func ExtractMultiPartUploadIDFrom(response *http.Response) (string, error) {
 		return "", errors.New("upload ID was empty")
 	}
 	return initiateMultipartUploadResult.UploadID, nil
-}
\ No newline at end of file
+}
+
+//ReplicateRequest makes a copy of the provided request
+func ReplicateRequest(ctx context.Context, request *http.Request, bodyBytes []byte) (*http.Request, error) {
```

Comment:
Move original body read here and report errors if needed

Reply:
Moving the original body here would require reading the body each time we wan't to replicate the request, i.e. for each storage call.